### PR TITLE
VMware utils functions: additional logs & raise NoPermission exceptions

### PR DESCRIPTION
**Pull request by alexbleotu**
### What does this PR do?

Logs VMWare exceptions and raises correct messages when vim.fault.NoPermission errors are caught

### Tests written?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/utils/vmware.py [hunk 1]
```diff
@@ -242,8 +242,8 @@ def _get_service_instance(host, username, password, protocol,
                     b64token=token,
                     mechanism=mechanism)
             else:
+                log.error(exc)
```
Comment (cachedout) [inline salt/utils/vmware.py]: This should probably be `log.exception`

File: salt/utils/vmware.py [hunk 2]
```diff
@@ -262,8 +262,8 @@ def _get_service_instance(host, username, password, protocol,
                         mechanism=mechanism
                     )
                 except Exception as exc:
+                    log.error(exc)
```
Comment (cachedout) [inline salt/utils/vmware.py]: same here wrt log.exception

**Review by cachedout — CHANGES_REQUESTED** [review]
There are a number of cases here where we could use `log.exception` instead of `log.error`. I've highlighted two but there are more.

File: salt/utils/vmware.py (hunk 1 above)
Comment (alexbleotu) [inline salt/utils/vmware.py]: Didn't know about this - thanks Mike!

File: salt/utils/vmware.py (hunk 2 above)
Comment (alexbleotu) [inline salt/utils/vmware.py]: Done

**Comment by alexbleotu** [comment]
@cachedout I've done the changes. Can you please have a look?

**Review by cachedout — APPROVED** [review]

**Pull request merged**